### DISCUSSION TITLE
9주차 스터디 회고

### DISCUSSION THREAD

**Discussion by alanhakhyeonsong** [Next-Level-Study/effective-java-study#27] [스터디 회고]
한 줄 회고와 건의사항을 말씀해주세요:)

**Comment by JeongJaew0n** [comment]
@sdh46 2주 연속발표 수고하셨습니다 ㅎㅎ
이번 주는 자바의 함수형 프로그래밍을 알게되어서 좋았습니다.
@alanhakhyeonsong 알려준 덕분에 자바에도 클로저가 있다는걸 알게 되었습니다!

**Reply by alanhakhyeonsong** [reply]
감사합니다^^

**Comment by sdh46** [comment]
람다와 스트림을 써보긴 했지만 정확한 개념을 인지하고 쓰진 않았는데 이번 기회에 좀 더 잘 알 수 있어서 좋습니다. 다음 주에는 실습 비중을 좀 더 늘려봐야겠습니다

**Comment by alanhakhyeonsong** [comment]
Java의 람다, 스트림을 그냥 사용하는 수준으로 코드를 싸질러놨는데 제가 알고 이렇게 코드를 작성했는지에 대해 반성하는 타임입니다.
TypeScript로 코드를 작성하다보면 함수를 값으로 쓰고 함수형 프로그래밍이 Java보단 자유롭게 사용할 수 있다보니 이렇게 코드를 작성하는 걸 Java의 람다/스트림을 사용할 때 적용해보면 정말 좋을 것 같단 생각이 들거든요.

각 언어의 좋은 점들을 최대한 활용할 수 있도록 습관을 들여봐야겠습니다.

**Comment by opgumo3** [comment]
처음 람다를 접했을 때는 이게 뭔 소리고 왜 보기 좋은지 몰랐었는데 (아마 익숙하지 않아서 그랬던 것 같습니다..)
다른 언어를 배우게 되면서, 이번 장을 공부하는 게 좀 수월했던 것 같습니다.
자바에서 함수형 인터페이스를 익숙해질 수 있도록 더 사용해봐야겠습니다!!

**Reply by alanhakhyeonsong** [reply]
동감합니다~

**Comment by SuJeong68** [comment]
람다와 스트림에 대해 공부했지만 제네릭과 클로저에 대해 생각하게 되는 시간이었습니다. 부족하지 않았나 싶네요. 더 실습해봐야겠습니다!

**Reply by alanhakhyeonsong** [reply]
제네릭을 생각하는 건 일단 메서드의 시그니처부터 생각해보고 클래스로 확장해나가면서 이해해보면 쉬울거에요. 클로저는 Java에서 딱히 언급이 안되는 내용이라 생소했을겁니다~

**Comment by narangd0** [comment]
마지막에 언급된 내용처럼, 그냥 IDLE 이 고치라고 해서 써왔던 것들에 대해 배울 수 있는 시간이어서 유익했다
@FunctionalInterface 의 사용처들을 직접 찾아보면 좋을 것 같다

**Reply by alanhakhyeonsong** [reply]
ㅋㅋㅋㅋ인정합니다
